controllers/releaseplanadmission/adapter_test.go

```diff
@@ -0,0 +1,272 @@
+
+		It("should overwrite previously matched ReleasePlans", func() {
+			adapter.releasePlanAdmission.Status.ReleasePlans = []v1alpha1.MatchedReleasePlan{
+				{Name: "rp-two", Active: false},
+				{Name: "foo", Active: false},
+			}
+			secondReleasePlan := releasePlan.DeepCopy()
+			secondReleasePlan.Name = "rp-two"
+
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlansContextKey,
+					Resource: &v1alpha1.ReleasePlanList{
+						Items: []v1alpha1.ReleasePlan{
+							*releasePlan,
+							*secondReleasePlan,
+						},
+					},
+				},
+			})
+
+			result, err := adapter.EnsureMatchingInformationIsSet()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.releasePlanAdmission.Status.ReleasePlans).To(HaveLen(2))
+			for i := range adapter.releasePlanAdmission.Status.Conditions {
+				if adapter.releasePlanAdmission.Status.Conditions[i].Type == "Matched" {
+					Expect(adapter.releasePlanAdmission.Status.Conditions[i].Status).To(Equal(metav1.ConditionTrue))
+				}
+			}
+			for _, releasePlan := range adapter.releasePlanAdmission.Status.ReleasePlans {
+				Expect(releasePlan.Name).NotTo(Equal("foo"))
+			}
+		})
+
+		It("should update the condition time if only the auto-release label on a ReleasePlan changes", func() {
+			testReleasePlan := releasePlan.DeepCopy()
+			testReleasePlan.Labels = map[string]string{}
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlansContextKey,
+					Resource: &v1alpha1.ReleasePlanList{
+						Items: []v1alpha1.ReleasePlan{
+							*testReleasePlan,
+						},
+					},
+				},
+			})
+
+			adapter.EnsureMatchingInformationIsSet()
+			condition := meta.FindStatusCondition(adapter.releasePlanAdmission.Status.Conditions, "Matched")
+			lastTransitionTime := condition.LastTransitionTime
+			time.Sleep(1 * time.Second)
+
+			testReleasePlan.Labels[metadata.AutoReleaseLabel] = "true"
+			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ReleasePlansContextKey,
+					Resource: &v1alpha1.ReleasePlanList{
+						Items: []v1alpha1.ReleasePlan{
+							*testReleasePlan,
+						},
+					},
+				},
+			})
+
+			adapter.EnsureMatchingInformationIsSet()
+			condition = meta.FindStatusCondition(adapter.releasePlanAdmission.Status.Conditions, "Matched")
+			Expect(condition.LastTransitionTime).NotTo(Equal(lastTransitionTime))
+			Expect(adapter.releasePlanAdmission.Status.ReleasePlans).To(HaveLen(1))
+			Expect(adapter.releasePlanAdmission.Status.ReleasePlans[0].Active).To(BeTrue())
+		})
+
+		/*It("should not update the lastTransitionTime in the condition if the matched ReleasePlans haven't changed", func() {
+			// this test fails because we don't get the releasePlanAdmission from the cluster, so although
+			// the patch command is not executed, we still see the modifications to it - TODO
```

Comment:
@davidmogar what are your thoughts on this?

Reply:
We will have a mock client in the toolkit. I  need to work on that for Kruno. The mock client would fix this, isn't it?